dlib/test/dnn.cpp

```diff
@@ -2875,6 +2875,131 @@ namespace
 
 // ----------------------------------------------------------------------------------------
 
```

Comment:
Note that the new tests would fail with the previous `loss.h`.

I guess it is a matter of specification whether the tests themselves make sense. But at least in my world, they do. :)